routers/web/auth/auth.go

```diff
@@ -155,6 +173,11 @@ func SignIn(ctx *context.Context) {
 		return
 	}
 
+	// Check if authentication is forced to OAuth
+	if checkForceOAuth(ctx) {
+		return
+	}
+
 	orderedOAuth2Names, oauth2Providers, err := oauth2.GetActiveOAuth2Providers()
```

Comment:
This is great! Thank you so much!! Are you able to re-use this `GetActiveOAuth2Providers` call, to reduce the duplication of DB calls?

Reply:
I tried that but sadly the map of Providers that `GetActiveOAuth2Providers` returns is a different object that doesn't contain all the Source attributes, including my ForceOAuth parameter